extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdNetworkDoubleclickImpl} from '../amp-ad-network-doubleclick-impl';
+import {AmpAdUIHandler} from '../../../amp-ad/0.1/amp-ad-ui'; // eslint-disable-line no-unused-vars
+import {
+  AmpAdXOriginIframeHandler,    // eslint-disable-line no-unused-vars
+} from '../../../amp-ad/0.1/amp-ad-xorigin-iframe-handler';
+import {base64UrlDecodeToBytes} from '../../../../src/utils/base64';
+import {utf8Encode} from '../../../../src/utils/bytes';
+import * as sinon from 'sinon';
+
+describe('amp-ad-network-doubleclick-impl', () => {
+
+  let sandbox;
+  let doubleclickImpl;
+  let doubleclickImplElem;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    doubleclickImplElem = document.createElement('amp-ad');
```

Comment:
If you're going to do `document.createElement`, worth doing it inside an isolated iframe, I think.  Can also use  `src/dom#createElementWithAttributes` as shorthand.  Try nesting the following inside the outer `describes.sandboxed`:
```javascript
  describes.fakeWin('some descriptive name', {}, env => {
    let win;
    let doc;
    beforeEach(() => {
      win = env.win;
      doc = win.document;
      doubleclickImplElem = createElementWithAttributes(doc, 'amp-ad', {
        type: 'doubleclick',
        data-ad-client: 'adsense',
        width: /* ??? */,
        height:  /* ??? */,
      });
      // Do rest of beforeEach initialization.
    });
    // Test cases and so on.
  });